### ISSUE TITLE
3.4 doesnt support transparent controls

### ISSUE THREAD

**Issue by ahrbe1** [godotengine/godot#52482]
### Godot version

3.4.beta4

### System information

Windows 10

### Issue description

It appears that the TextEdit background_color cannot be set to fully transparent in a theme. When this setting is applied, the TextEdit still displays the default blue/purple color. This appears to be the case in both the background color Theme Override, as well as if Color(0, 0, 0, 0) is present in the theme itself. I believe this worked in 3.3.3 stable official

### Steps to reproduce

1. Create a TextEdit
2. In the inspector, set "Syntax Highlight"=On (required for background_color override to work)
3. Set Theme Overrides -> Colors -> Background Color to something fully transparent
4. TextEdit shows the blue/purple color instead of what's behind it.

### Minimal reproduction project

[transparency.zip](https://github.com/godotengine/godot/files/7124955/transparency.zip)

**Comment by ahrbe1** [comment]
FWIW, I was eventually able to get around this by writing a custom shader. Though the current behavior was surprising/non-intuitive.

**Comment by pycbouh** [comment]
Do you want to remove the existing background from the TextEdit? If so, you need to set up its styleboxes, namely the "normal" one and the "focus" one. Background color is used for something else.

**Comment by ahrbe1** [comment]
Ok. I'll try that. Thanks.

**Comment by ahrbe1** [comment]
Ok. So I can confirm that the suggested solution resolves my issue.

Do you think it warrants further discussion about the intended behavior when selecting transparent for the background color? I do find it to be a somewhat confusing behavior, but I'll leave it to you all to decide if this ticket should be closed or not. Thanks.

**Comment by pycbouh** [comment]
Looking at the code, I'm not entirely sure why it's needed. It just draws on top of the stylebox if it has some color to it (alpha over 0.01). In stable it also checks for syntax highlighting being enabled indeed. Because of that I think it may be a hack to expose it to the editor settings for the script editor without touching the styling of the `TextEdit` node. Though it can still be done with custom styleboxes, removing the current confusion.

@Paulb23 what do you think?

**Comment by ahrbe1** [comment]
Is that used by the theme loader?

**Comment by pycbouh** [comment]
> Is that used by the theme loader?

What do you mean? What theme loader?

**Comment by ahrbe1** [comment]
I mean the editor syntax themes. They seem to contain a background_color, e.g. 

https://github.com/godotengine/godot-syntax-themes

```
[color_theme]
background_color="ff242424"
completion_background_color="ff343434"
...
```

Maybe that's unrelated?

**Comment by pycbouh** [comment]
Ah, yes, that's what I've said here:

> I think it may be a hack to expose it to the editor settings for the script editor without touching the styling of the TextEdit node. Though it can still be done with custom styleboxes, removing the current confusion.

**Comment by ahrbe1** [comment]
Ah. Ok. Right on.

**Comment by Paulb23** [comment]
Yeah if I remember correctly it was originally only designed for the editor. 

We could remove it, shouldn't cause any issues apart from maybe usability, as its bit more work to generate 3 styleboxes rather then changing a colour.

**Comment by pycbouh** [comment]
We can still expose it to the settings as a color, but instead of just passing it to the script editor as a color we can generate 3 styleboxes (or rather, copy the existing ones and change their bg color).

Basically, just a bit of work around this line:

```
theme->set_color("background_color", "CodeEdit", EDITOR_GET("text_editor/theme/highlighting/background_color"));
```

**Comment by kitbdev** [comment]
- The `background_color` for editor text themes can be made transparent since #61454

The TextEdit `background_color` is no longer in use by the editor at all, and I don't think it has any benefits over just using the StyleBox, so I think we should mark it as deprecated and tell users to use the StyleBox instead.
